% Use with CC terms.
% Adrin Jalali - 2013
%

\documentclass{beamer}
%\setbeamertemplate{navigation symbols}[frame number]

\usepackage{beamerthemeshadow}
\usepackage[absolute,overlay]{textpos}
\usepackage{graphics}
\usepackage{colortbl}
\usepackage{xcolor}
\usepackage[absolute,overlay]{textpos}
\usepackage{sidecap}
\usepackage[labelformat=empty]{caption}
\usepackage{booktabs}

\setbeamercolor{framesource}{fg=gray}
\setbeamerfont{framesource}{size=\tiny}
\setbeamertemplate{footline}[frame number]
\beamertemplatenavigationsymbolsempty

\newcommand{\source}[1]{\begin{textblock*}{4cm}(8.7cm,8.6cm)
    \begin{beamercolorbox}[ht=0.5cm,right]{framesource}
        \usebeamerfont{framesource}\usebeamercolor[fg]{framesource} {#1}
    \end{beamercolorbox}
\end{textblock*}}

\newcommand{\mycite}[1]{\begin{textblock*}{4cm}(8.7cm,8.6cm)
    \begin{beamercolorbox}[ht=0.5cm,right]{framesource}
        \usebeamerfont{framesource}\usebeamercolor[fg]{framesource} {#1}
    \end{beamercolorbox}
\end{textblock*}}

\definecolor{pathwaynode}{RGB}{255,150,50}
\definecolor{independentnode}{RGB}{255,255,50}
\newcommand{\boz}{\cellcolor{pathwaynode}}
\newcommand{\ghool}{\cellcolor{independentnode}}

\begin{document}
\title[Gene Expression, PPI Network, Classification]{Analyzing How Protein Interaction Networks Improve Classification Performance in Gene Expression Data Analysis}
\author[Adrin Jalali]{Adrin Jalali \\ \small{Supervised by: Nico Pfeifer}}
\institute{MPI for Informatics}
\date{\today} 
\titlegraphic{\includegraphics[width=4cm]{mpi-logo}}

\begin{frame}
\titlepage
\end{frame}

\begin{frame}
  \frametitle{Overview}
  \begin{enumerate}
  \item Introduction and data
  \item Formulate the approach - incorporate PPI-Network in SVM
  \item Results
  \item Work in progress - graph kernels
  \end{enumerate}
\end{frame}

\section{Intro}
\begin{frame}
  \frametitle{Microarray Gene Expression}
  \begin{columns}
    \begin{column}{0.5\textwidth}
      \begin{figure}
        \includegraphics[width=0.9\textwidth]{Affymetrix-microarray}
      \end{figure}
    \end{column}
    \begin{column}{0.5\textwidth}
      \begin{figure}
        \includegraphics[width=0.8\textwidth]{Microarray-schema}
      \end{figure}
    \end{column}
  \end{columns}
  \source{en.wikipedia.org}
  \note{http://en.wikipedia.org/wiki/DNA_microarray}
\end{frame}

\begin{frame}
  \frametitle{van 't Veer Breast-Cancer Data}
  \begin{columns}
    \begin{column}{0.5\textwidth}
      \begin{figure}
        \includegraphics[width=1\textwidth]{vantveer-summary}
      \end{figure}
    \end{column}
    \begin{column}{0.5\textwidth}
      \begin{enumerate}
      \item Input: Gene expression data
      \item Output: Prognosis (Poor vs. Good), Metastases
      \item Goal: Classify and find important genes
      \item Issue: Hard to classify due to huge number of features (genes) compared to number of samples ($\sim 22000 \gg 98$)
      \end{enumerate}
    \end{column}
  \end{columns}
  \mycite{Laura J. van 't Veer et.al. Nature, (2002)}
\note{a, Two-dimensional presentation of transcript ratios for 98 breast tumours. There were 4,968 significant genes across the group. Each row represents a tumour and each column a single gene. As shown in the colour bar, red indicates upregulation, green downregulation, black no change, and grey no data available. The yellow line marks the subdivision into two dominant tumour clusters. b, Selected clinical data for the 98 patients in a: BRCA1 germline mutation carrier (or sporadic patient), ER expression, tumour grade 3 (versus grade 1 and 2), lymphocytic infiltrate, angioinvasion, and metastasis status. White indicates positive, black negative and grey denotes tumours derived from BRCA1 germline carriers who were excluded from the metastasis evaluation. The cluster below the yellow line consists of 36 tumours, of which 34 are ER negative (total 39 ER-negative) and 16 are carriers of the BRCA1 mutation (total 18). c, Enlarged portion from a containing a group of genes that co-regulate with the ER- gene (ESR1). Each gene is labelled by its gene name or accession number from GenBank. Contig ESTs ending with RC are reverse-complementary of the named contig EST. d, Enlarged portion from a containing a group of co-regulated genes that are the molecular reflection of extensive lymphocytic infiltrate, and comprise a set of genes expressed in T and B cells. (Gene annotation as in c.)}
\note{http://www.nature.com/nature/journal/v415/n6871/full/415530a.html}
\end{frame}

\begin{frame}
  \frametitle{Yeast Protein Interaction Network}
\begin{figure}
\includegraphics[width=0.6\textwidth]{yeastProteinInteractionNetwork}
\caption{\tiny The colour of a node indicates the phenotypic effect of removing the corresponding protein ({\color{red}red} = lethal, {\color{green}green} = non-lethal, {\color{orange}orange} = slow growth, {\color{yellow}yellow} = unknown)}
\end{figure}
\note{http://www.nature.com/nrg/journal/v5/n2/fig_tab/nrg1272_F2.html}
\source{A. Barab√°si, Z. Oltvai, Nature Reviews Genetics, (2004)}
\end{frame}


\section{Formulation} 

\begin{frame}
  \begin{columns}
    \begin{column}{0.5\textwidth}
      \begin{figure}
        \includegraphics[width=1\textwidth]{NICK-paper}
      \end{figure}
    \end{column}
    \begin{column}{0.5\textwidth}
      \begin{enumerate}
      \item It's shown:
        \begin{itemize}
        \item Co-expressed genes tend to be close in the PPI-Network.
        \item Exploit this fact to modify the SVM objective function - called NICK
        \end{itemize}
      \item What can be done:
        \begin{itemize}
        \item Reverse engineer the learned machine to extract important genes after using the network information.
        \end{itemize}
      \end{enumerate}
    \end{column}
  \end{columns}
\end{frame}


\begin{frame}
  \frametitle{NICK}
  \begin{columns}
    \begin{column}{0.5\textwidth}
      
      \begin{block}{\tiny{1. SVM modified objective function}}
        \tiny
        \begin{center}
          $\min_{\mathbf{w}, w_0}\left\{\frac{1}{2}\|\mathbf{w}\|^2 + \frac{1}{2}\beta\sum_{(j,k)\in E}(w_j-w_k)^2\right\}$
        \end{center}
        s.t.:
        \begin{center}
          $\forall i \in \{1,\cdots,n\} : (\mathbf{w}\mathbf{x}_i+w_0)y_i\geq 1$
        \end{center}
      \end{block}
      
      \begin{block}{\tiny{3. Dual to Primal}}
        \tiny
        \begin{center}
          $\mathbf{w} = (\mathbf{I} + \beta \mathbf{B})^{-1} \sum_{i = 1}^n \alpha_i y_i \mathbf{x}_i$
        \end{center}
      \end{block}
    \end{column}
    
    \begin{column}{0.5\textwidth}
      \begin{block}{\tiny{2. Dual Problem}}
        \tiny
        \begin{center}
          \begin{align*}
            &\max_\alpha\left\{\sum_{i=1}^n\alpha_i-\frac{1}{2}\sum_{i=1}^n\sum_{j=1}^n\alpha_i\alpha_j y_i y_j (\mathbf{x}_i^T\mathbf{L})(\mathbf{L}^T\mathbf{x}_j)\right\}\\
            &\mathbf{L}\mathbf{L}^T=(\mathbf{I}+\beta \mathbf{B})^{-1}\\
            \text{s.t.: }&\\
            &\forall i \in \{1,\cdots,n\}: \sum_{i=1}^n\alpha_iy_i=0\\
            &\forall i \in \{1,\cdots,n\}: \alpha_i \geq 0 \\
            &\text{Laplacian matrix:}\\  & \mathbf{B} = \mathbf{D} - \mathbf{A}
          \end{align*}
        \end{center}
      \end{block}
    \end{column}
  \end{columns}
  \mycite{Ofer Lavi, et.al., Journal of Computational Biology, (2012)}
\end{frame}

\begin{frame}
  \frametitle{NICK Performance Summary}
  \begin{figure}
    \includegraphics[width=0.8\textwidth]{NICK-perfs}
  \end{figure}
  \mycite{Ofer Lavi, et.al., Journal of Computational Biology, (2012)}
  \note{http://online.liebertpub.com/doi/full/10.1089/cmb.2012.0065}
\end{frame}

\begin{frame}
  \frametitle{Verify What Has To Be Done}
  \begin{enumerate}
  \item To be done: extract important genes.
  \item There is no gold standard for it.
  \item Synthesized data for the purpose of method verification.
  \end{enumerate}
\end{frame}

\begin{frame}
\frametitle{Synthesize Data}
\begin{enumerate}
\item A random graph (PPI-Network)
\item Signal nodes (genes): \[ f(n) = \left\{ 
  \begin{array}{l l}
    N(-\mu, 1) & \quad \text{if $n$ is in class $1$}\\
    N(\mu, 1) & \quad \text{if $n$ is in class $2$}
  \end{array} \right.\]
\item Random nodes (non-informative genes): \[f(n) = N(0, 1) \]
\item Pathway: 2, 3, or 4 connected signal nodes.
\end{enumerate}
\end{frame}

\begin{frame}
  \frametitle{Synthesized Data}
\only<1-2>{
  \begin{figure}
    \includegraphics[width=0.8\textwidth]{synthesized}
  \end{figure}
}
\only<2-3>{
  \begin{textblock*}{\paperwidth}(0.4\textwidth,0.2\textheight)
    \raggedright
  \begin{figure}
    \includegraphics[angle=270,width=0.2\textwidth]{signal-nodes}
  \end{figure}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(-0.4\textwidth,0.2\textheight)
    \raggedright
  \begin{figure}
    \includegraphics[angle=270,width=0.2\textwidth]{random-nodes}
  \end{figure}
  \end{textblock*}
}
\only<3>{
  \begin{figure}
    \includegraphics[width=0.8\textwidth]{synthesized-easy}
  \end{figure}
  \begin{textblock*}{\paperwidth}(0.1\textwidth,\textheight)
    \raggedright
    \tiny{{\color{blue}Blue}: random gene, {\color{orange}Orange}: Signal node being a member of a pathway of signal nodes, {\color{yellow}Yellow}: A lonely signal node}
  \end{textblock*}
}
\end{frame}

\begin{frame}
  \frametitle{Extract Important Genes}
  \begin{itemize}
  \item Solve SVM problem for original and transformed data.
  \item Calculate $\mathbf{w}$ for both models.
  \item Compute for each pair of nodes, for each model: 
    \begin{block}{}
      $Score(i, j) = \frac{|w_i| + |w_j|}{2} \times e^{max\left(d_G(i, j), 1\right)}$
    \end{block}
  \item Report pairs with highest scores for both trained models.
  \end{itemize}
\end{frame}

\section{Results}
\begin{frame}
  \frametitle{Synthesized Data Easy Scenario}
  \begin{figure}
    \includegraphics[width=0.8\textwidth]{synthesized-easy}
  \end{figure}
  \begin{textblock*}{\paperwidth}(0.01\textwidth,0.25\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\toprule
\multicolumn{2}{c}{Original} \\ 
\midrule \hline
\boz X196   &  \boz X196  \\ \hline
X53   &  X53  \\ \hline
X233   &  X233  \\ \hline
X39   &  X39  \\ \hline
\ghool X88   &  \ghool X88  \\ \hline
\boz X196   &  \boz X133  \\ \hline
\boz X116   &  \boz X116  \\ \hline
X127   &  X127  \\ \hline
X197   &  X197  \\ \hline
X127   &  X148  \\ \hline
X148   &  X148  \\ \hline
\boz X150   &  \boz X150  \\ \hline
X148   &  X273  \\ \hline
\boz X116   &  \boz X133  \\ \hline
X160   &  X160  \\ \hline
\ghool X96   &  \ghool X96  \\ \hline
X95   &  X95  \\ \hline
X273   &  X273  \\ \hline
\ghool X88   &  X115  \\ \hline
X40   &  X40  \\ \hline
X53   &  X8  \\ \hline
X53   &  X164  \\ \hline
X195   &  X195  \\ \hline
X56   &  X56  \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(1\textwidth,0.25\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\toprule
\multicolumn{2}{c}{Transformed} \\ 
\midrule \hline
\boz X196   &  \boz X196  \\ \hline
X233   &  X233  \\ \hline
\boz X196   &  \boz X133  \\ \hline
\boz X133   &  \boz X133  \\ \hline
\boz X133   &  \boz X116  \\ \hline
\boz X116   &  \boz X116  \\ \hline
X95   &  X95  \\ \hline
\boz X240   &  \boz X240  \\ \hline
X39   &  X39  \\ \hline
\boz X240   &  \boz X243  \\ \hline
X59   &  X59  \\ \hline
X106   &  X106  \\ \hline
\boz X243   &  \boz X243  \\ \hline
X106   &  X168  \\ \hline
X114   &  X114  \\ \hline
X168   &  X168  \\ \hline
\boz X243   &  \boz X150  \\ \hline
X56   &  X56  \\ \hline
X39   &  X47  \\ \hline
X298   &  X298  \\ \hline
\boz X150   &  \boz X150  \\ \hline
X247   &  X247  \\ \hline
\boz X125   &  \boz X125  \\ \hline
X83   &  X83  \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(0.4\textwidth,1\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
AUC (Original): & 60.6 \\ \hline
AUC (Transformed): & 62.4 \\ \hline
      wc p-value (paired): & 5.669e-09 \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
\end{frame}

\begin{frame}
  \frametitle{Synthesized Data Medium Scenario}
  \begin{figure}
    \includegraphics[width=0.8\textwidth]{synthesized-medium}
  \end{figure}
  \begin{textblock*}{\paperwidth}(0.01\textwidth,0.25\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\toprule
\multicolumn{2}{c}{Original} \\ 
\midrule \hline
\boz X190   &  \boz X190  \\ \hline
X104   &  X104  \\ \hline
X233   &  X233  \\ \hline
\boz X190   &  \boz X272  \\ \hline
X277   &  X277  \\ \hline
\ghool X88   &  \ghool X88  \\ \hline
\boz X190   &  X127  \\ \hline
X165   &  X165  \\ \hline
\boz X272   &  \boz X272  \\ \hline
\boz X272   &  \boz X22  \\ \hline
X106   &  X106  \\ \hline
X165   &  \ghool X96  \\ \hline
\boz X150   &  \boz X150  \\ \hline
X250   &  X250  \\ \hline
\ghool X88   &  X215  \\ \hline
\boz X22   &  \boz X22  \\ \hline
X51   &  X51  \\ \hline
X28   &  X28  \\ \hline
X73   &  X73  \\ \hline
X35   &  X35  \\ \hline
X162   &  X162  \\ \hline
\boz X113   &  \boz X113  \\ \hline
X112   &  X112  \\ \hline
X277   &  X102  \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(1\textwidth,0.25\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\toprule
\multicolumn{2}{c}{Transformed} \\ 
\midrule \hline
X233   &  X233  \\ \hline
\boz X190   &  \boz X190  \\ \hline
X112   &  X112  \\ \hline
\boz X240   &  \boz X240  \\ \hline
\boz X190   &  \boz X272  \\ \hline
\boz X240   &  \boz X243  \\ \hline
X86   &  X86  \\ \hline
\boz X243   &  \boz X243  \\ \hline
\boz X243   &  \boz X150  \\ \hline
\boz X190   &  X127  \\ \hline
\boz X150   &  \boz X150  \\ \hline
\boz X272   &  \boz X272  \\ \hline
X246   &  X246  \\ \hline
X298   &  X298  \\ \hline
X106   &  X106  \\ \hline
\boz X125   &  \boz X125  \\ \hline
X35   &  X35  \\ \hline
\boz X125   &  \boz X82  \\ \hline
X247   &  X247  \\ \hline
\boz X272   &  X69  \\ \hline
\boz X272   &  \boz X22  \\ \hline
\boz X82   &  \boz X82  \\ \hline
X100   &  X100  \\ \hline
\ghool X257   &  \ghool X257  \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(0.4\textwidth,1\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
AUC (Original): & 60.1 \\ \hline
AUC (Transformed): & 61.5 \\ \hline
      wc p-value (paired): & 1.383e-06 \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
\end{frame}

\begin{frame}
  \frametitle{Synthesized Data Hard Scenario}
  \begin{figure}
    \includegraphics[width=0.8\textwidth]{synthesized-hard}
  \end{figure}
  \begin{textblock*}{\paperwidth}(0.01\textwidth,0.25\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\toprule
\multicolumn{2}{c}{Original} \\ 
\midrule \hline
\boz X190   &  \boz X190  \\ \hline
X101   &  X101  \\ \hline
X233   &  X233  \\ \hline
\boz X190   &  \boz X272  \\ \hline
\ghool X88   &  \ghool X88  \\ \hline
X297   &  X297  \\ \hline
\boz X190   &  X127  \\ \hline
X93   &  X93  \\ \hline
X26   &  X26  \\ \hline
\boz X138   &  \boz X138  \\ \hline
\boz X272   &  \boz X272  \\ \hline
\boz X272   &  \boz X22  \\ \hline
X101   &  X41  \\ \hline
X123   &  X123  \\ \hline
\boz X22   &  \boz X22  \\ \hline
X101   &  X198  \\ \hline
X146   &  X146  \\ \hline
\boz X228   &  \boz X228  \\ \hline
X278   &  X278  \\ \hline
X72   &  X72  \\ \hline
\ghool X88   &  X115  \\ \hline
\ghool X96   &  \ghool X96  \\ \hline
\boz X148   &  \boz X148  \\ \hline
X112   &  X112  \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(1\textwidth,0.25\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\toprule
\multicolumn{2}{c}{Transformed} \\ 
\midrule \hline
X233   &  X233  \\ \hline
\boz X190   &  \boz X190  \\ \hline
X112   &  X112  \\ \hline
\boz X190   &  \boz X272  \\ \hline
X86   &  X86  \\ \hline
\boz X190   &  X127  \\ \hline
\boz X272   &  \boz X272  \\ \hline
\boz X272   &  X205  \\ \hline
X205   &  X205  \\ \hline
X146   &  X146  \\ \hline
X146   &  X68  \\ \hline
X68   &  X68  \\ \hline
X298   &  X298  \\ \hline
\boz X272   &  \boz X22  \\ \hline
X90   &  X90  \\ \hline
X127   &  X127  \\ \hline
X100   &  X100  \\ \hline
\boz X272   &  X69  \\ \hline
X297   &  X297  \\ \hline
X72   &  X72  \\ \hline
X127   &  \boz X148  \\ \hline
X155   &  X155  \\ \hline
X247   &  X247  \\ \hline
X196   &  X196  \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(0.4\textwidth,1\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
      wc p-value (paired): & 8.151e-13 \\ \hline
AUC (Original): & 60.2 \\ \hline
AUC (Transformed): & 62.5 \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
\end{frame}

\begin{frame}
  \frametitle{van 't Veer}
  \begin{textblock*}{\paperwidth}(0.01\textwidth,0.25\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\toprule
\multicolumn{2}{c}{Original} \\ 
\midrule \hline
X85453  &  X85453 \\ \hline 
X85453  &  X92140 \\ \hline 
X6605  &  X6605 \\ \hline 
X56886  &  X56886 \\ \hline 
X10640  &  X10640 \\ \hline 
X8817  &  X8817 \\ \hline 
X56894  &  X56894 \\ \hline 
X6605  &  X332 \\ \hline 
X5733  &  X5733 \\ \hline 
X57758  &  X57758 \\ \hline 
X7532  &  X7532 \\ \hline 
X51  &  X51 \\ \hline 
X7566  &  X7566 \\ \hline 
X3267  &  X3267 \\ \hline 
X89953  &  X89953 \\ \hline 
X5713  &  X5713 \\ \hline 
X5193  &  X5193 \\ \hline 
X5365  &  X5365 \\ \hline 
X10874  &  X10874 \\ \hline 
X5982  &  X5982 \\ \hline 
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(0.93\textwidth,0.25\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\toprule
\multicolumn{2}{c}{Transformed} \\ 
\midrule \hline
X9917  &  X9917 \\ \hline 
X84279  &  X84279 \\ \hline 
X197370  &  X197370 \\ \hline 
X51143  &  X51143 \\ \hline 
X58475  &  X58475 \\ \hline 
X55585  &  X55585 \\ \hline 
X25949  &  X25949 \\ \hline 
X54892  &  X54892 \\ \hline 
X126695  &  X126695 \\ \hline 
X57168  &  X57168 \\ \hline 
X10456  &  X10456 \\ \hline 
X148223  &  X148223 \\ \hline 
X9742  &  X9742 \\ \hline 
X253558  &  X253558 \\ \hline 
X342527  &  X342527 \\ \hline 
X10175  &  X10175 \\ \hline 
X83930  &  X83930 \\ \hline 
X57035  &  X57035 \\ \hline 
X145482  &  X145482 \\ \hline 
X57465  &  X57465 \\ \hline 
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(0.4\textwidth,1\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
      wc p-value (paired): & 0.006 \\ \hline
AUC (Original): & 72.9 \\ \hline
AUC (Transformed): & 73.6 \\ \hline
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
\only<2>{
  \begin{textblock*}{\paperwidth}(0.26\textwidth,0.3\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\cellcolor[gray]{0.8} Node & \cellcolor[gray]{0.8} Degree \\ \hline
X85453  &  12 \\ \hline 
X6605  &  98 \\ \hline 
X56886  &  26 \\ \hline 
X10640  &  16 \\ \hline 
X8817  &  152 \\ \hline 
X56894  &  28 \\ \hline 
X5733  &  150 \\ \hline 
X57758  &  8 \\ \hline 
X7532  &  86 \\ \hline 
X51  &  172 \\ \hline 
X7566  &  16 \\ \hline 
X3267  &  56 \\ \hline 
X89953  &  4 \\ \hline 
X5713  &  126 \\ \hline 
X5193  &  32 \\ \hline 
X5365  &  70 \\ \hline 
X10874  &  132 \\ \hline 
X5982  &  172 \\ \hline 
X92140  &  20 \\ \hline 
X332  &  328 \\ \hline 
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
  \begin{textblock*}{\paperwidth}(0.68\textwidth,0.3\textheight)
    \raggedright 
    \tiny
    \begin{tabular}{| c c |}
      \hline
\cellcolor[gray]{0.8} Node & \cellcolor[gray]{0.8} Degree \\ \hline
X9917  &  0 \\ \hline 
X84279  &  0 \\ \hline 
X197370  &  0 \\ \hline 
X51143  &  0 \\ \hline 
X58475  &  0 \\ \hline 
X55585  &  0 \\ \hline 
X25949  &  0 \\ \hline 
X54892  &  0 \\ \hline 
X126695  &  0 \\ \hline 
X57168  &  0 \\ \hline 
X10456  &  0 \\ \hline 
X148223  &  0 \\ \hline 
X9742  &  0 \\ \hline 
X253558  &  0 \\ \hline 
X342527  &  0 \\ \hline 
X10175  &  0 \\ \hline 
X83930  &  0 \\ \hline 
X57035  &  0 \\ \hline 
X145482  &  0 \\ \hline 
X57465  &  0 \\ \hline 
    \end{tabular}
    \hspace{.5em}
  \end{textblock*}
}
\end{frame}

\section{Work in progress}
\begin{frame}
  \frametitle{Idea}
  \begin{enumerate}
\footnotesize
  \item Estimate gene expression value probability distributions for samples of class A. \pause
  \item For each sample in class A and B:
    \begin{enumerate}
\scriptsize
    \item Extract abnormal genes according to above estimated distributions.
    \item Extract the part of PPI network induced by extracted genes (almost)
    \end{enumerate}\pause
  \item Use a graph kernel for labeled graphs to classify extracted graphs. 
%  \item Extract common sub-graphs from individual graphs that seem to be helping the classification. \pause
  \end{enumerate}
\end{frame}

\begin{frame}
  \frametitle{Graph Kernel}
\only<1>{
      \begin{block}{\tiny{SVM Dual Problem}}
        \tiny
        \begin{center}
          \begin{align*}
            &\max_\alpha\left\{\sum_{i=1}^n\alpha_i-\frac{1}{2}\sum_{i=1}^n\sum_{j=1}^n\alpha_i\alpha_j y_i y_j \langle \mathbf{x}_i . \mathbf{x}_j \rangle \right\}\\
            \text{s.t.: }&\\
            &\forall i \in \{1,\cdots,n\}: \sum_{i=1}^n\alpha_iy_i=0\\
            &\forall i \in \{1,\cdots,n\}: \alpha_i \geq 0 \\
          \end{align*}
        \end{center}
      \end{block}
}
\only<2>{
      \begin{block}{\tiny{SVM Dual Problem}}
        \tiny
        \begin{center}
          \begin{align*}
            &\max_\alpha\left\{\sum_{i=1}^n\alpha_i-\frac{1}{2}\sum_{i=1}^n\sum_{j=1}^n\alpha_i\alpha_j y_i y_j \langle \phi(\mathbf{x}_i) . \phi(\mathbf{x}_j) \rangle \right\}\\
            \text{s.t.: }&\\
            &\forall i \in \{1,\cdots,n\}: \sum_{i=1}^n\alpha_iy_i=0\\
            &\forall i \in \{1,\cdots,n\}: \alpha_i \geq 0 \\
          \end{align*}
        \end{center}
      \end{block}
}
\only<3>{
      \begin{block}{\tiny{SVM Dual Problem}}
        \tiny
        \begin{center}
          \begin{align*}
            &\max_\alpha\left\{\sum_{i=1}^n\alpha_i-\frac{1}{2}\sum_{i=1}^n\sum_{j=1}^n\alpha_i\alpha_j y_i y_j K(\mathbf{x}_i, \mathbf{x}_j)\right\}\\
            \text{s.t.: }&\\
            &\forall i \in \{1,\cdots,n\}: \sum_{i=1}^n\alpha_iy_i=0\\
            &\forall i \in \{1,\cdots,n\}: \alpha_i \geq 0 \\
          \end{align*}
        \end{center}
      \end{block}
}
\only<4>{
  \begin{itemize}
  \item Random Walks ($A^k$: number of walks of length $k$)
  \item Sub-Tree Kernels
  \item Shortest Paths Kernels (1-shortest path, k-shortest paths)
  \item Graphlet Kernels (Isomorphism proved for $n = k + 1 , n \le 11$)
  \item Laplacian Matrix - Eigenvalues ($\text{Isomorph}(G_1, G_2) \iff \exists P: B(G_1) = P^tB(G_2)P$), \\ number of connected components = number of $\lambda_i = 0$
  \end{itemize}
}
\end{frame}

\begin{frame}
  \frametitle{Idea}
  \begin{enumerate}
\footnotesize
  \item Estimate gene expression value probability distributions for samples of class A. 
  \item For each sample in class A and B:
    \begin{enumerate}
\scriptsize
    \item Extract abnormal genes according to above estimated distributions.
    \item Extract the part of PPI network induced by extracted genes (almost)
    \end{enumerate}
  \item Use a graph kernel for labeled graphs to classify extracted graphs.
%  \item Extract common sub-graphs from individual graphs that seem to be helping the classification. \pause
  \end{enumerate}
  \begin{textblock*}{\paperwidth}(0.4\textwidth,0.85\textheight)
    \raggedright
  \begin{figure}
    \includegraphics[angle=270,width=0.2\textwidth]{signal-nodes}
  \end{figure}
  \end{textblock*}
\end{frame}

\begin{frame}
  \frametitle{Weighted Idea}
\only<1>{
  \begin{figure}
    \includegraphics[width=1\textwidth]{weighted-1}
    \caption{Class A member}
  \end{figure}
  \begin{textblock*}{0.2\paperwidth}(0.9\textwidth,0.3\textheight)
    \raggedright
\tiny
{\color{blue}Blue}: Low Expressed\\
{\color{red}Red}: High Expressed
  \end{textblock*}
}
\only<2>{
  \begin{figure}
    \includegraphics[width=1\textwidth]{weighted-2}
    \caption{Class B member}
  \end{figure}
  \begin{textblock*}{0.2\paperwidth}(0.9\textwidth,0.3\textheight)
    \raggedright
\tiny
{\color{blue}Blue}: Low Expressed\\
{\color{red}Red}: High Expressed
  \end{textblock*}
}
\end{frame}

\begin{frame}
  \frametitle{Next ...}
  \begin{itemize}
  \item Same structured graphs for each sample - even same edge weights.
  \item Most kernels:
    \begin{itemize}
    \item Detect structural differences on graphs.
    \item Assume nodes are not labeled.
    \end{itemize}
  \item Design/Find/Change a kernel for our graphs.
  \item Reverse engineer the kernel on support vectors to detect common substructures on them.
  \end{itemize}
\end{frame}

\begin{frame}[plain]
\frametitle{Acknowledgment}
\begin{itemize}
\item Thomas Lengauer
\item Nico Pfeifer
\item Sarvesh Nimkubh
\item Nora Speicher
\item Anna Feldmann
\end{itemize}
\end{frame}

\begin{frame}[plain]
\frametitle{Finished!}
  \begin{center}
    \Huge{Thank You!}\\
    Questions?
  \end{center}
\end{frame}

\end{document}
